### ISSUE TITLE
[CLI]: Pytorch lightning + wandb - nested metrics not being logged correctly?

### ISSUE THREAD

**Issue by kdu4108** [wandb/wandb#5549]
### Describe the bug

Hi, when I initiate a run with Pytorch Lightning's [WandbLogger](https://lightning.ai/docs/pytorch/latest/extensions/generated/lightning.pytorch.loggers.WandbLogger.html), I find that nested metrics are unexpectedly no longer logged to the "summary" section of a run. 

Example: 

```python
from pytorch_lightning.loggers import WandbLogger
logger = WandbLogger(
    project="debug-summary-metrics",
    config={"p1": 1, "p2": 3},
    tags=["debug", "lightning"],
)
wandb.log({"m1": 10})
wandb.log({"train": {"m2": 10}})
wandb.finish()
```
results in
![image](https://github.com/wandb/wandb/assets/13302561/d98d5fae-5855-4a5d-8794-4e1e18b853a2)
which is missing the `train.m2` metric.

Curiously enough, the charts do show that the `train.m2` nested metric exists, so it's being logged somewhere, but it's not visible in the summary section.

Therefore, when I try to retrieve the nested metrics using the Run API (`run.summary`), the nested metrics aren't there. 

Can you please help me figure out how I can retrieve these nested metrics? Thank you!

### Environment

WandB version: '0.13.5'

OS: Ubuntu 22.04

Python version: 3.9.15

Versions of relevant libraries: 2.0.0

**Comment by rsanandres-wandb** [comment]
Hello @kdu4108 !

Could you send me a link to your workspace/project? I ran your code snippet and was able to see `train.m2` on my project. Let me know if there is a difference in my implementation and yours that may be the reason why the `wandb.summary` is different.

[Overview of my Test](https://wandb.ai/raphael-sanandres/pytorch-logging/runs/f3l0yydd/overview?workspace=user-raphael-sanandres)
[Code used](https://wandb.ai/raphael-sanandres/pytorch-logging/runs/f3l0yydd/files/code/tester_folder/tester.py)

**Comment by rsanandres-wandb** [comment]
After some debugging, I was able to replicate this issue. I will be making a bug report on this. It looks like that I added a `wandb.init()` which let me log the `train.m2` so adding a `wandb.init()` should allow you to see `train.m2`. However, just initiating a project with `WandbLogger` does cause the bug that you see.

**Comment by kdu4108** [comment]
Great, thank you! I will make sure I run `wandb.init()` going forward. However, is there a way to access the metrics for previous runs?

**Comment by rsanandres-wandb** [comment]
Unfortunately, `wandb` did not log the correct `train.m2` value anywhere. However, we do save a git remote url and git commit id in the `wandb-metadata.json` and you may be able to grab the results there. (For example, you can find it [here](https://wandb.ai/ethz-rycolab/debug-summary-metrics/runs/3smqx4ba/files/wandb-metadata.json) in the project that you sent before).
